### DISCUSSION TITLE
Project Structure

### DISCUSSION THREAD

**Discussion by Benanna2019** [hyper63/atlas#75] [Q&A] [answered]
Stages of a Product (Initial client conversations, product design, development phase, production, etc). 

Do you know of any good ways to structure or go about approaching taking on a client/project and then the different phases of a project? (Clarification: Basically I have been thinking a lot about 'architecture' in the sense of having the initial client conversations, designing the product, building the product and pushing the product to production and it would be most helpful to have a structure (which is what I am looking for)).

**Comment by twilson63** [comment, marked as answer]
Here is an article that kind of approaches this subject at a high level, I would love for you to read it and let me know your thoughts.
https://blog.hyper63.com/apis/

There are so many options that lay across a spectrum of
idea -> prototype -> beta -> v1 -> v2 -> v3 ...

In these times, you want to create fast-flow, which means you are in a state of constant iteration that iteration consists of three core task types:
	new features
	change requests
	chores or maintenance or technical debt

As you approach each phase of the product, following the Technology Adoption curve - https://ondigitalmarketing.com/learn/odm/foundations/5-customer-segments-technology-adoption/ - you have different decisions to deal with along the way, from product to architecture. But the most important is about the team, the team needs to function as the architects and navigate the product forward as a decisive group, own each decision and manage each failure.

**Reply by Benanna2019** [reply]
The blog article that you referenced was incredibly helpful. Thinking through the benefits of this structure is exciting. To be able to keep all the business logic separate and not have it scattered across your whole application seems like it makes for a more innovative process, especially as it pertains to maintenance of the app after production.
